starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,488 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else {
+    return num2;
+  }
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(arrOfWords) {
+  let longestWord = [];
+  let counter = 0;
+
+  if (arrOfWords.length === 0) {
+    return null;
+  } else {
+    for (let i = 0; i < arrOfWords.length; i++) {
+      if (arrOfWords[i].length > counter) {
+        counter = arrOfWords[i].length;
+        longestWord.unshift(arrOfWords[i]);
+      }
+    }
+
+    return longestWord[0];
+  }
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(arrOfNumbers) {
+  let sum = 0;
+
+  for (let i = 0; i < arrOfNumbers.length; i++) {
+    sum += arrOfNumbers[i];
+  }
+
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(arrOfNumbers) {
+  if (arrOfNumbers.length === 0) {
+    return null;
+  } else {
+    let average = sumArray(arrOfNumbers) / arrOfNumbers.length;
+    return average;
```

Comment:
Perfect. you could also do it in one line -> return sumArray(arrOfNumbers)/arrOfNumbers